README.md

```diff
@@ -37,185 +41,198 @@ Simple parsing for node.js.
 
 ```js
 expression -> number "+" number {%
-    function (data, location, reject) {
-        return ["sum", data[0], data[2]];
-    }
+    (data, location, reject) => ({
+        type: "sum",
+        args: [data[0], data[2]]
+    })
 %}
 ```
 
-`data` is an array whose elements match the nonterminals in order. The
-postprocessor `id` returns the first token in the match (literally
-`function(data) {return data[0];}`).
+The rule above will parse `5+10` into `{ type: "sum", args: [5, 10] }`.
+
+The postprocessor can be any function. It will be passed three arguments:
 
-`location` is the index at which that rule was found. Retaining this
-information in a syntax tree is useful if you're writing an interpreter and
-want to give fancy error messages for runtime errors.
+- `data: Array` - array with the parsed parts of the rule. It will always be an array, even if there's only one part. If a rule contains nonterminals with their own postprocessors, the respective parts will already be transformed.
+- `location: number` - the index (zero-based) at which the rule match starts. It's useful to retain this information in the syntax tree if you're writing an interpreter.
+- `reject: Object` - return this object to signal that this rule doesn't actually match. This allows you to restrict or conditionally support language features.
 
-If, after examining the data, you want to force the rule to fail anyway, return
-`reject`. An example of this is allowing a variable name to be a word that is
-not a string:
+Remember that a postprocessor is scoped to a single rule, not the whole nonterminal. If a nonterminal has multiple alternative rules, each of them can have its own postprocessor:
 
 ```js
-variable -> word {%
-    function(data, location, reject) {
-        if (KEYWORDS.indexOf(data[0]) === -1) {
-            return data[0]; // It's a valid name
-        } else {
-            return reject;  // It's a keyword, so reject it
-        }
-    }
-%}
+expression ->
+      number "+" number {% ([first, _, second]) => first + second %}
+    | number "-" number {% ([first, _, second]) => first - second %}
+    | number "*" number {% ([first, _, second]) => first * second %}
+    | number "/" number {% ([first, _, second]) => first / second %}
 ```
 
-You can write your postprocessors in CoffeeScript by adding `@preprocessor
-coffee` to the top of your file. Similarly, you can write them in TypeScript by
-adding `@preprocessor typescript` to the top of your file. If you would like to
-support a different postprocessor language, feel free to file a PR!
+There're several built-in postprocessors for the most common scenarios:
+
+- `id` - returns the first element of the `data` array. Useful for single-element arrays: `foo -> bar {% id %}`
+- `arrpush` - joins an array and the last element. Useful for recursive definitions of sequences: `foo -> bar | foo bar {% arrpush %}`
```

Comment:
Let's omit `arrpush` here, because it's frequently misunderstood, causing perf issues.